src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11956,6 +12025,91 @@ private ExpressionSyntax ParseNewExpression()
+            var openBracket = this.EatToken(SyntaxKind.OpenBracketToken);
+
+            var list = _pool.AllocateSeparated<CollectionElementSyntax>();
+
+            if (this.CurrentToken.Kind != SyntaxKind.CloseBracketToken)
+            {
+tryAgain:
+                if (IsPossibleCollectionCreationExpression() || this.CurrentToken.Kind == SyntaxKind.CommaToken)
+                {
+                    list.Add(this.ParseCollectionElement());
+
+                    int lastTokenPosition = -1;
+                    while (IsMakingProgress(ref lastTokenPosition))
+                    {
+                        if (this.CurrentToken.Kind == SyntaxKind.CloseBracketToken)
+                        {
+                            break;
+                        }
+                        else if (IsPossibleCollectionCreationExpression() || this.CurrentToken.Kind == SyntaxKind.CommaToken)
+                        {
+                            list.AddSeparator(this.EatToken(SyntaxKind.CommaToken));
+
+                            // check for exit case after legal trailing comma
+                            if (this.CurrentToken.Kind == SyntaxKind.CloseBraceToken)
+                            {
+                                break;
+                            }
+                            else if (!IsPossibleCollectionCreationExpression())
+                            {
+                                goto tryAgain;
+                            }
+
+                            list.Add(this.ParseCollectionElement());
+                            continue;
+                        }
+                        else if (SkipBadCollectionCreationExpressionTokens(ref openBracket, list, SyntaxKind.CommaToken) == PostSkipAction.Abort)
+                        {
+                            break;
+                        }
+                    }
+                }
+                else if (SkipBadCollectionCreationExpressionTokens(ref openBracket, list, SyntaxKind.CommaToken) == PostSkipAction.Continue)
+                {
+                    goto tryAgain;
+                }
+            }
+
+            return _syntaxFactory.CollectionCreationExpression(
+                openBracket,
+                _pool.ToListAndFree(list),
+                this.EatToken(SyntaxKind.CloseBracketToken));
+
+            PostSkipAction SkipBadCollectionCreationExpressionTokens(ref SyntaxToken openBracket, SeparatedSyntaxListBuilder<CollectionElementSyntax> list, SyntaxKind expected)
+            {
+                return this.SkipBadSeparatedListTokensWithExpectedKind(ref openBracket, list,
+                    p => p.CurrentToken.Kind != SyntaxKind.CommaToken && !p.IsPossibleCollectionCreationExpression(),
+                    p => p.CurrentToken.Kind == SyntaxKind.CloseBracketToken || p.IsTerminator(),
+                    expected);
+            }
+        }
+
+        private bool IsPossibleCollectionCreationExpression()
```

Comment:
>IsPossibleCollectionCreationExpression [](http://example.com/codeflow?start=21&length=38)

Should the be named `IsPossibleCollectionElement()`?

Reply:
er yes.  oops.